### PR TITLE
Login form

### PR DESCRIPTION
**Pull request by tkhoma**
Added login functionality

### CODE REVIEW

File: src/components/Header/Header.js [hunk 1]
```diff
@@ -14,16 +14,15 @@ class Header extends Component {
     }
 
     componentDidMount() {
-        if (document.location.pathname !== '/') {
-            this.props.dispatch(headerActions.show());
-        }
-        window.addEventListener('scroll', _.throttle(this.handleScroll), 500);
+        window.addEventListener('scroll', _.throttle(this.handleScroll, 500));
```
Comment (oziniak) [inline src/components/Header/Header.js]: be sure to `removeEventListener` in `componentDidUnmount`

File: src/components/Login/LoginForm.js [hunk 2]
```diff
@@ -1,23 +1,105 @@
...
+    }
+
+    handleChange(e) {
+        const { name, value } = e.target;
+        this.setState({ [name]: value });
+    }
+
+    onSubmit(e) {
+        e.preventDefault();
+        this.setState({ submitted: true });
```
Comment (oziniak) [inline src/components/Login/LoginForm.js]: should this be changed only once user logging in is `success`?

File: src/components/Login/LoginForm.js [hunk 3]
```diff
@@ -1,23 +1,105 @@
...
-        )
+        const { username, password, submitted } = this.state;
+
+        if (this.props.isLoggedIn) {
+            return (
+                <PrivatePage />
+            )
+        } else {
+            return (
+                <div className="container">
```
Comment (oziniak) [inline src/components/Login/LoginForm.js]: better to extract this to a separate component

File: src/components/Login/LoginForm.js [hunk 4]
```diff
@@ -1,23 +1,105 @@
...
 
     render() {
-        return (
-            <div>
-                Hello from Login Form
-            </div>
-        )
+        const { username, password, submitted } = this.state;
+
+        if (this.props.isLoggedIn) {
```
Comment (oziniak) [inline src/components/Login/LoginForm.js]: should this check exist in render, if you redirect in `componentDidUpdate` anyway?

File: src/actions/userActions.js [hunk 5]
```diff
@@ -1,56 +1,29 @@
...
-export function getTransactions(id) {
-  return {
-    [CALL_API]: {
-      types: [types.TRANSACTIONS_REQUEST, types.TRANSACTIONS_SUCCESS, types.TRANSACTIONS_FAILURE],
-      endpoint: `get_transactions`,
-    },
-  };
+function logout() {
+    userService.logout();
+    return { type: userConstants.LOGOUT };
```
Comment (oziniak) [inline src/actions/userActions.js:27]: is `userService.logout()` synchronous?

File: src/actions/userActions.js [hunk 6]
```diff
@@ -1,56 +1,29 @@
...
- * @returns {{}}
- */
-export function getBalance() {
-  return {
-    [CALL_API]: {
-      types: [types.USER_BALANCE_REQUEST, types.USER_BALANCE_SUCCESS, types.USER_BALANCE_FAILURE],
-      endpoint: `get_balance`,
-    },
-  };
+    function success(user) { return { type: LOGIN_SUCCESS, user, isLoggedIn: true } }
```
Comment (oziniak) [inline src/actions/userActions.js]: please move to the upper scope

File: src/components/Main.js [hunk 7]
```diff
@@ -1,15 +1,48 @@
...
-        </Switch>
-    </main>
-);
+class Main extends Component {
+    constructor(props) {
+        super(props);
 
-export default Main
+        const { dispatch } = this.props;
+        this.props.history.listen((location, action) => {
```
Comment (oziniak) [inline src/components/Main.js]: should this listener be removed once the component is destroyed?

File: src/services/userService.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+            sessionStorage.setItem('user', JSON.stringify(user));
+            return user;
+        });
+}
+
+function handleResponse(response) {
+    return response.text().then(text => {
+        const data = text && JSON.parse(text);
+        if (!response.ok) {
+            console.log(data.message);
```
Comment (oziniak) [inline src/services/userService.js]: please remvoe

**Review by oziniak — APPROVED** [review]

File: src/actions/userActions.js (hunk 5 above)
Comment (tkhoma) [reply src/actions/userActions.js:27]: logout is not implemented yet. will be done in next pull request

File: src/components/Login/LoginForm.js (hunk 4 above)
Comment (tkhoma) [reply src/components/Login/LoginForm.js]: I redirect in `componentDidUpdate` because I don't want to show login page for already logged in user

**Review by oziniak — APPROVED** [review]

**Pull request merged**